### PR TITLE
[chromecast-api] Add types

### PR DESCRIPTION
**Pull request by SteffoSpieler**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## chromecast-api ([<kbd>unpkg</kbd>](https://unpkg.com/browse/chromecast-api@latest/))
was missing the following properties:
1. once
1. on
1. errorMonitor
1. captureRejectionSymbol
1. captureRejections

<details>
<summary>as well as these 1 other properties...</summary>
<p>EventEmitter</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ca90b4a9314c80fc8a99e831e135beddd8eb28aa
</p>

File: types/chromecast-api/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for chromecast-api 0.4
+// Project: https://github.com/alxhotel/chromecast-api
+// Definitions by: SteffoSpieler <https://github.com/SteffoSpieler>
+//                 daniel0611 <https://github.com/daniel0611>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import Client from "./lib/client";
```
Comment (peterblazejewicz) [inline types/chromecast-api/index.d.ts:7]:
```suggestion
import Client = require("./lib/client");
```
IMO this is all CJS module, old school:
https://github.com/alxhotel/chromecast-api/blob/master/index.js

File: types/chromecast-api/lib/client.d.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { EventEmitter } from "events";
+import Device from "./device";
+
+/**
+ * Initialize the client to start searching for chromecast devices.
+ */
+export default class Client extends EventEmitter {
```
Comment (peterblazejewicz) [inline types/chromecast-api/lib/client.d.ts:7]:
```suggestion
declare class Client extends EventEmitter {
```

File: types/chromecast-api/lib/client.d.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    constructor();
```
Comment (peterblazejewicz) [inline types/chromecast-api/lib/client.d.ts:8]:
```suggestion
```
it's always there for parameterless ctor

File: types/chromecast-api/lib/client.d.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+     */
+    querySSDP(): void;
+
+    /**
+     * Trigger the mDNS and SSDP search again. Warning: the device event will trigger again (it might return the same device).
+     */
+    update(): void;
+
+    destroy(): void;
+}
```
Comment (peterblazejewicz) [inline types/chromecast-api/lib/client.d.ts:36]:
```suggestion
}

export = Client;
```

File: types/chromecast-api/lib/device.d.ts [hunk 5]
```diff
@@ -0,0 +1,207 @@
+import { EventEmitter } from "events";
+
+export interface DeviceOptions {
```
Comment (peterblazejewicz) [inline types/chromecast-api/lib/device.d.ts:3]: this unfortunately, have to be rewritte,

so you should have, AFAIK: 
```ts
namespace Device {
    interface DeviceOptions {
    ...
    interface MediaResource {
    ...
}

declare class Device ...

export = Device;
````
https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-class-d-ts.html

File: types/chromecast-api/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true
```
Comment (peterblazejewicz) [inline types/chromecast-api/tsconfig.json:18]:
```suggestion
        "forceConsistentCasingInFileNames": true
```
https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

**Comment by peterblazejewicz** [comment]
the last bit:
```ts
// Definitions by: SteffoSpieler <https://github.com/SteffoSpieler>
//                 daniel0611 <https://github.com/daniel0611>
```
we keep a rule, adding real people into the owner should involve explicit ACK from that person, as DT will send notification to package owner during PR lifecycles. Notice that docs mention 'yourself' when describing adding owners:
https://github.com/DefinitelyTyped/DefinitelyTyped#definition-owners

**Comment by daniel0611** [comment]
@peterblazejewicz  I've co-authored this PR with @SteffoSpieler and do want to be a definition owner for it. Is this comment enough or should I add myself in another PR?

Thanks

**Comment by peterblazejewicz** [comment]
This is OK

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@SteffoSpieler thanks!

**Comment by SteffoSpieler** [comment]
Ready to merge

**Pull request merged**